apps/application/flow/step_node/speech_to_text_step_node/impl/base_speech_to_text_node.py

```diff
@@ -37,25 +39,34 @@ def process_audio_item(audio_item, model):
                 temp_mp3_path = temp_amr_file.name
             any_to_mp3(temp_file_path, temp_mp3_path)
             try:
-                return split_and_transcribe(temp_mp3_path, model)
+                transcription = split_and_transcribe(temp_mp3_path, model)
+                return {file.file_name: transcription}
             finally:
                 os.remove(temp_file_path)
                 os.remove(temp_mp3_path)
 
         def process_audio_items(audio_list, model):
             with ThreadPoolExecutor(max_workers=5) as executor:
                 results = list(executor.map(lambda item: process_audio_item(item, model), audio_list))
-            return '\n\n'.join(results)
+            return results
 
         result = process_audio_items(audio_list, stt_model)
-        return NodeResult({'answer': result, 'result': result}, {})
+        content = []
+        result_content = []
+        for item in result:
+            for key, value in item.items():
+                content.append(f'### {key}\n{value}')
+                result_content.append(value)
+        return NodeResult({'answer': '\n'.join(result_content), 'result': '\n'.join(result_content),
+                           'content': content}, {})
 
     def get_details(self, index: int, **kwargs):
         return {
             'name': self.node.properties.get('stepName'),
             "index": index,
             'run_time': self.context.get('run_time'),
             'answer': self.context.get('answer'),
+            'content': self.context.get('content'),
             'type': self.node.type,
             'status': self.status,
             'err_message': self.err_message,
```

Comment:
The provided code has several issues and could be improved:

### Issues:
1. **Variable Naming**: The variable name `stt_model` is not defined. It should be replaced with the actual model instance used for speech-to-text conversion.

2. **Splitter Consistency**: The line `splitter = '\n`-----------------------------------`\n';` uses double backslashes `\\`, which will cause issues because it won't escape the newlines correctly. You should use a single backslash `\`.

3. **Content Handling**: The logic for handling content seems incorrect. Instead of adding all items into a single content list, a nested loop approach would allow you to add each item's details under the corresponding file.

4. **Error Handling**: Ensure that any errors related to saving or processing files are caught and handled appropriately in your error handling mechanism.

5. **Node Result Structure**: The structure of `NodeResult` might need adjustment based on how your application expects results to be formatted. For example, using dictionaries instead of lists can make more sense if each key-value pair represents a different piece of data.

### Improvements:
1. **Fix Variable Name**:
   ```python
-    # Replace stt_model here with the actual model instance you're using
```

2. **Correct Backslashes**:
   ```python
+    splitter = r'\n-------------------------\n'
   ```

3. **Optimized Content Handling**:
   Change the way content is collected:
   ```python
   def process_audio_items(audio_list, model):
       result_with_content = {}
       
       with ThreadPoolExecutor(max_workers=5) as executor:
           futures = [executor.submit(process_audio_item, item, model) for item in audio_list]
           
           for future in futures:
               audio_data, transcription = future.result()
               file_name = audio_data["file_name"]
               result_with_content[file_name] = {
                   "transcription": transcription,
                   "content": [
                       f"### {file_name}\n{transcription}"
                   ]
               }
   
       return NodeResult({"data": result_with_content})
   ```

4. **Consistent Error Handling**:
   Implement proper error logging and handling mechanisms within the `process_audio_item` method and other parts of the function where file operations occur.

5. **Review Node Result Structure**:
   If needed, adjust `NodeResult` to better fit your data needs. Consider returning dictionaries with clear keys indicating what information each part represents (e.g., `answer`, `results`, etc.)

Here’s an updated version incorporating some of these improvements:

```python
from joblib.pool import ThreadPoolExecutor
from typing import Dict

from .models import File
from setting.models_provider.tools import get_model_instance_by_model_user_id
import os

# Assuming this is the correct model instance for speech recognition
stt_model = get_model_instance_by_model_user_id(user_id)

def process_audio_item(audio_item):
    temp_mp3_path = f"/tmp/temp_{audio_item.file.filename}.mp3"
    try:
        any_to_mp3(audio_item.file.path, temp_mp3_path)
        
        transcription_result = split_and_transcribe(temp_mp3_path, stt_model)
        transcriptions = [{file.file_name: entry} for file, entries in transcription_result.items()]
        return {'file': audio_item.file.filename, 'entries': transcriptions}
    finally:
        os.remove(temp_mp3_path)

class BaseSpeechToTextNode ISpeechToTextNode):

    ...

    def process_audio_items(self, audio_list):
        # Use dictionary comprehension to store results directly
        results_dicts = []
        with ThreadPoolExecutor(max_workers=5) as executor:
            futures = [executor.submit(process_audio_item, audio_item) for audio_item in audio_list]
            
            for future in futures:
                audio_data_dict = future.result()
                results_dicts.extend(audio_data_dict['entries'])

        flattened_results = [{'file': d['file'], 'entry': e} for sub_d in results_dicts for e in sub_d['entries']]
        return NodeResult({'flattenedResults': flattened_results})

    ...
```